python/python/bystro/proteomics/handler.py

```diff
@@ -0,0 +1,362 @@
+    job_data: SaveJobData,
+    search_conf: dict,
+    max_query_size: int = 10_000,
+    max_slices=1024,
+    keep_alive=ONE_DAY,
+) -> pd.DataFrame:
+    assert isinstance(max_query_size, int), type(max_query_size)
+    header = bytes("\t".join(job_data.fieldNames) + "\n", encoding="utf-8")
+
+    search_client_args = gather_opensearch_args(search_conf)
+    client = OpenSearch(**search_client_args)
+    delimiters = get_delimiters()
+
+    query = _preprocess_query(job_data.queryBody)
+    num_slices = _get_num_slices(client, job_data.indexName, max_query_size, max_slices, query)
+    point_in_time = client.create_point_in_time(  # type: ignore[attr-defined]
+        index=job_data.indexName, params={"keep_alive": keep_alive}
+    )
+    pit_id = point_in_time["pit_id"]
+    query["pit"] = {"id": pit_id}
+    query["size"] = max_query_size
+    remote_queries = []
+    for slice_id in range(num_slices):
+        slice_query = query.copy()
+        if num_slices > 1:
+            # Slice queries require max > 1
+            slice_query["slice"] = {"id": slice_id, "max": num_slices}
+        remote_query = _process_query_pure.remote(  # type: ignore[call-arg]
+            query_args={"body": slice_query},
+            search_client_args=search_client_args,
+            field_names=job_data.fieldNames,
+            delimiters=delimiters,
+        )
+        remote_queries.append(remote_query)
+    results_processed = ray.get(remote_queries)
+    client.delete_point_in_time(body={"pit_id": pit_id})  # type: ignore
+    if RAY_ERROR in results_processed:
+        failing_idx = results_processed.index(RAY_ERROR)
+        err_msg = (
+            f"Encountered Ray error while processing query {failing_idx} of {len(results_processed)}"
+        )
+        raise OSError(err_msg)
+
+    payload = [header, *results_processed]
+    return _process_payload(payload, delimiters)
+
+
+def _process_payload(payload: list[bytes], delimiters: dict[str, str]) -> pd.DataFrame:
+    payload_str = "".join([byte_str.decode("utf-8") for byte_str in payload])
+    samples_and_genes_df = pd.read_csv(io.StringIO(payload_str), sep="\t")
+    assert all(
+        samples_and_genes_df.columns == ["discordant", "homozygotes", "heterozygotes", "refSeq.name2"]
+    )
+    samples_and_genes_df = samples_and_genes_df.drop("discordant", axis=1)
+    samples_and_genes_df = samples_and_genes_df.rename({"refSeq.name2": "gene_name"}, axis=1)
+    output = []
+    for _i, row in samples_and_genes_df.iterrows():
+        gene_names = deduplicate(split_gene_names(row.gene_name, delimiters))
+        homozygotes = row.homozygotes.split("|") if row.homozygotes != "!" else []
+        heterozygotes = row.heterozygotes.split("|") if row.heterozygotes != "!" else []
+        for gene_name in gene_names:
+            for heterozygote in heterozygotes:
+                output.append((heterozygote, gene_name, 1))
+            for homozygote in homozygotes:
```

Comment:
We probably also want to store missingGenos. Missing values are themselves interesting. Without knowing what is missing you can't know if the individual is reference. Researchers may care about folks who had ambiguous (missing) calls because they're non-reference